### PR TITLE
Update energy tests

### PR DESCRIPTION
**Pull request by mattwthompson**
This is a follow-on to #4 with just a couple of substantive changes

* The water dimer evaluation helper in `smirnoff-plugins` makes incorrect assumptions about atom/particle ordering, so I just ported it over to here with the right order.
* For relatively simple energy evaluations, Interchange's OpenMM function seems to work fine

### CODE REVIEW

File: deforcefields/tests/test_deforcefields.py [hunk 1]
```diff
@@ -89,3 +89,91 @@ def test_fails_unsupported_chemistry():
...
+    ff = ForceField("de-force-1.0.2.offxml", load_plugins=True)
+
+    found_energies = evaluate_water_energy_at_distances(
+        force_field=ff,
+        distances=[2, 3, 4],
+    )
+    ref_energies = [1005.0846252441406, 44.696786403656006, 10.453390896320343]
+
+    for found, ref in zip(found_energies, ref_energies):
+        assert found.m_as(kj_mol) == pytest.approx(ref, rel=1e-3)
```
Comment (mattwthompson) [inline deforcefields/tests/test_deforcefields.py]: Note this is a fairly loose tolerance; I was getting errors like these if I used the defaults. How exactly these energies have drifted slightly from their artisan hand-crafted values a couple of years back, I'm not sure 


```
$ python -m pytest -v deforcefields/tests/test_deforcefields.py::test_energy_sites | grep "E    "
E       assert 1005.0861476907567 == 1005.0846252441406 ± 1.0e-03
E
E         comparison failed
E         Obtained: 1005.0861476907567
E         Expected: 1005.0846252441406 ± 1.0e-03
E       assert 44.69819545912324 == 44.696786403656006 ± 4.5e-05
E
E         comparison failed
E         Obtained: 44.69819545912324
E         Expected: 44.696786403656006 ± 4.5e-05
E       assert 10.454626091527247 == 10.453390896320343 ± 1.0e-05
E
E         comparison failed
E         Obtained: 10.454626091527247
E         Expected: 10.453390896320343 ± 1.0e-05

Comment (jthorton) [reply deforcefields/tests/test_deforcefields.py]: I think the original values were calculated using the toolkit pre-interchange but maybe its time to do them by hand to double check, for now though this is a great match!

**Review by jthorton — APPROVED** [review]
Fantastic work @mattwthompson, hopefully, this should be the end oV-site-site issues!

**Comment by mattwthompson** [comment]
Hopefully!

If everything's up to date, could we get a release (maybe 1.0.2?) to get this rolled out to the "new" infrastructure? I'm happy to do the feedstock PR, but I'm not going to tag a release on one of your projects 😅

**Pull request merged**